volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,254 @@
+    # which the original plugin used to do it.
+
+    # I don't think it's worth including the sessions method either because both the original psxview plugin
+    # and Volatility3's sessions plugin begin with the list of processes found by PsList.
+    # The original psxview plugin's session code essentially just filters the pslist for processes
+    # whose session ID is not None. I've matched this in my code, but again, it doesn't seem worth including.
+
+    # Lastly, I've omitted the pspcid method because I could not for the life of me get it to work. I saved the
+    # code I do have from it, and will happily share it if anyone else wants to add it.
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    valid_proc_name_chars = set(
+        string.ascii_lowercase + string.ascii_uppercase + "." + " "
+    )
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="info", component=info.Info, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="psscan", component=psscan.PsScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="thrdscan", component=thrdscan.ThrdScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="handles", component=handles.Handles, version=(1, 0, 0)
+            ),
+            requirements.BooleanRequirement(
+                name="physical-offsets",
+                description="List processes with phyiscall offsets instead of virtual offsets.",
+                optional=True,
+            ),
+        ]
+
+    def _proc_name_to_string(self, proc):
+        return proc.ImageFileName.cast(
+            "string", max_length=proc.ImageFileName.vol.count, errors="replace"
+        )
+
+    def _is_valid_proc_name(self, str):
+        for c in str:
+            if not c in self.valid_proc_name_chars:
+                return False
+        return True
+
+    def _filter_garbage_procs(self, proc_list):
+        return [
+            p
+            for p in proc_list
+            if p.is_valid() and self._is_valid_proc_name(self._proc_name_to_string(p))
+        ]
+
+    def _translate_offset(self, offset):
+        if not self.config["physical-offsets"]:
+            return offset
+
+        kernel = self.context.modules[self.config["kernel"]]
+        layer_name = kernel.layer_name
+
+        try:
+            _, _, offset, _, _ = list(
```

Comment:
If possible I prefer to show what the other values to make it easier for people that want to tinker (so `_original_offset, _original_length, offset, _length, _layer_name`, but I'm not gonna be a stickler about it...  5;)

Reply:
It's not assigning variables since they start with `_` (anymore than just using `_`) it would just save people needing to go elsewhere to get the information, but fair enough, as I said, not a showstopper.

Reply:
I didn't know about that syntax! Could be worth adding if only to showcase some obscure (only to me?) syntax

Reply:
Well, happy for you to add it before I merge?  I try to make sure it's used in most calls to mapping, just because it returns so many values and they're often discarded.  I'm having difficulty finding the official python reference, but it seems `_` is just a convention and gets treated just like any other variable, but we say that it won't get used (and type hinters know that, allowing the variable to be reused with different types without issue).  As I say, I just prefer having the name so people referring to the code can more clearly see what's going on.

Reply:
Just added it!